docs(upgrade-guide): add additional upgrade notes (#16800)
vuetifyjs/vuetify: packages/docs/src/pages/en/getting-started/upgrade-guide.md

- The variant naming scheme has changed slightly, it is now a single word instead of two. For example, `primary darken-1` is now `primary-darken-1`.
  - To use variant namings as value for `color` props, the variant you intend to use needs to be enabled in the theme under `theme.variations.colors`. e.g: `colors: ['primary']`
- Color classes have been renamed:
  - Backgrounds have a `bg-` prefix, for example `.primary` is now `.bg-primary`.
  - Text colors have a `text-` prefix, for example `.primary--text` is now `.text-primary`.
  - Variants are no longer a separate class, for example `.primary--text.text--darken-1` is now `.text-primary-darken-1`.
- The theme system now uses CSS variables internally, so `customProperties` is no longer required.
  - If you were using `customProperties` in v2, the naming scheme has changed from `--v-primary-base` to `--v-theme-primary`.
  - Custom properties are now also an rgb list instead of hex so `rgb()` or `rgba()` must be used to access them, for example `color: rgb(var(--v-theme-primary))` instead of `color: var(--v-primary-base)`.

> - Theme colors in the theme config are now nested inside a `color` property, e.g. `const myTheme = { theme: { themes: { light: { colors: { primary: '#ccc' } } } } }`

Reviewer: ```suggestion
- Theme colors in the theme config are now nested inside a `colors` property, e.g. `const myTheme = { theme: { themes: { light: { colors: { primary: '#ccc' } } } } }`
```

> - `$container-padding-x` is now 16px instead of 12px as in v2. You can replace it with `$spacer * 3` to get to the previous look

Reviewer: Not part of theming, maybe layout?

### SCSS

- `$headings` was merged with `$typography`: Access font-size of subtitle-2 with `map-get($typography, 'subtitle-2', 'size')`

> - If you imported variables from `~vuetify/src/styles/settings/_variables` in v2, you have to replace it with `~vuetify/_settings`

Reviewer: ```suggestion
- If you imported variables from `~vuetify/src/styles/settings/_variables` in v2, you have to replace it with `vuetify/settings`
```

> - Component variables that previously lived in e.g. `~/vuetify/src/components/VIcon/VIcon.sass` can now be imported from `~vuetify/_settings` directly too

Reviewer: ```suggestion
- Component variables that previously lived in e.g. `~/vuetify/src/components/VIcon/VIcon.sass` can now be imported from `vuetify/settings` directly too
```

- `$displayBreakpoints` no longer includes `{breakpoint}-only` variables (e.g. xs-only), use `@media #{map-get(v.$display-breakpoints, 'xs')}` instead

> - Replace `map-get(v.$transition, 'fast-in-fast-out)` with `$accelerated-easing`

Reviewer: ```suggestion
- The `$transition` map has been removed, replaced with individual `$standard-easing`, `$decelerated-easing`, `$accelerated-easing` variables.
```

> - `$vuetify.breakpoint` has been renamed to `$vuetify.display` and extended with [new properties](/features/display-and-platform/)

Reviewer: Not related to sass

Author: Moved to Theme section

Reviewer: Not really related to theme either. I guess layout is the closest section we already have for this

Author: moved

## Components

### General changes

- `value` prop has been replaced by `model-value` on components that support `v-model` usage.

>   - `v-btn`'s within `v-btn-toggle` still need a `value` prop though, do not rename

Reviewer: ```suggestion
  - Note that this does not apply to `value` used as a *selection value*, for example `v-btn` within `v-btn-toggle`.
```

- `@input` event has been replaced by `@update:model-value` on components that support `v-model` usage.
- `left` and `right` have been replaced by `start` and `end` respectively. This applies to utility classes too, for example `.border-r` is now `.border-e`.
- Size props `small` / `medium` / `large` etc. have been combined into a single `size` prop.
- `absolute` and `fixed` props have been combined into a single `position` prop.
- `top` / `bottom` / `left` / `right` props have been combined into a single `location` prop.
- `background-color` prop has been renamed to `bg-color`.

> - `dense` prop on components such as v-select, v-btn-toggle, v-alert, v-text-field, v-list and v-list-item has been changed to `density` prop with the variants `default`, `compact`, `comfortable`

Reviewer: ```suggestion
- `dense` prop on components such as v-select, v-btn-toggle, v-alert, v-text-field, v-list and v-list-item has been changed to `density` prop with the variants `default`, `comfortable`, `compact`
```

  - Allowed values are `'elevated'`, `'flat'`, `'tonal'`, `'outlined'`, `'text'`, or `'plain'`.
- `text` prop has new purpose. It represents the text content of the alert, if default slot is not used.

### v-btn/v-btn-toggle

- `active-class` prop has been renamed to `selected-class`.
- `fab` is no longer supported. If you just need a round button, use `icon` prop or apply a `.rounded-circle` class
- `flat` / `outlined` / `text` / `plain` props have been combined into a single `variant` prop.
- `depressed` has been renamed to `variant="flat"`.
- `retain-focus-on-click` has been removed, buttons use `:focus-visible` instead.
- `v-btn-toggle` needs `mandatory="force"` prop to achieve the same behaviour as `mandatory` prop in v2

> - `v-btn` has a new size `x-large`, which is why you have to down-shift previously used sizes by one to get the same size as in v2. E.g. `small` becomes `x-small`

Reviewer: xLarge existed in v2

Author: Can't reproduce exactly what the issues was. Removed.

### v-checkbox/v-radio/v-switch

- `input-value` prop has been renamed to `model-value`.
- `on-icon` and `off-icon` props have been renamed to `true-icon` and `false-icon`.
- `on-value` and `off-value` props have been renamed to `true-value` and `false-value`.
- `v-switch` now uses an input of type "checkbox", it does no longer have an `aria-checked` attribute to assert in tests.

> - `v-switch` no longer has `aria-role="switch"`, use `aria-role="checkbox"` to match in tests

Reviewer: I think there's probably way too many structural changes like this to be worth including here.

Author: Made a generic comment in the "General changes" section with v-switch as example

- `v-checkbox`'s label slot should no longer contain a `<label>` as it is already wrapped with one

### v-form

> - `validate()` now always returns a Promise that needs to be awaited. It also resolves on validation errors! Use `valid` and `errors` properties returned to determine form state

Reviewer: ```suggestion
- `validate()` now returns a [`Promise<FormValidationResult>`](/api/v-form/#exposed-validate) instead of a boolean. Await the promise then check `result.valid` to determine form state. 
```

- `ticks` has been renamed to `show-ticks`.
- `tick-labels` has been renamed to `ticks`.
- `vertical` has been renamed to `direction="vertical"`.

> - `v-slider`'s `step` prop is now 0 by default instead of 1 in v2

Reviewer: ```suggestion
- `step` default value is now 0 instead of 1.
```

### v-tabs

- `v-tab-item` has been removed, use `v-window-item`

### v-img

> - `contain` has been removed. It is the default behaviour now

Reviewer: ```suggestion
- `contain` has been removed and is now the default behaviour. Use `cover` to fill the entire container.
```

### v-menu

- `rounded` prop has been removed. Apply a rounded css class to the menu content element instead. e.g. `.rounded-te`
- `internal-activator` prop has been removed without replacement
- `offset-y` and `offset-x` props have been removed. Use `offset` prop instead
- `absolute` variant has been removed. For absolute positioning use css instead

> - `window.requestIdleCallback` and `window.cancelIdleCallback` need to be stubbed in tests, because `v-menu` uses them

Reviewer: This was only in 3.1.4
